### PR TITLE
feat: oidc implementation

### PR DESCRIPTION
**Pull request by thisyahlen-deriv**
This pull request includes several changes that enhance the login flow, add new components, and update dependencies. The most important changes include the introduction of a new login handling hook, the addition of a callback page, and updates to existing components to use the new login handling mechanism.

### Enhancements to login flow:

* [`src/hooks/useHandleLogin/index.tsx`](diffhunk://#diff-77f2c715c150077469ba59337440a261f88fab146e432bafbcf52679ac720342R1-R37): Introduced `useHandleLogin` hook to handle the new login flow using OIDC, which redirects users to the OAuth2 authorization page if OAuth2 is enabled, or falls back to the legacy OAuth URL otherwise.

### New components:

* `src/features/Callback/CallbackPage.tsx` and `src/features/Callback/index.ts`: Added `CallbackPage` component to handle successful sign-in and update login accounts. [[1]](diffhunk://#diff-3499a2994e8e41731b5942fd6814fa22e1f87c6374d6160d8957ff6725c4f347R1-R22) [[2]](diffhunk://#diff-0932d0c6630c69b7051deb288d4903716b2713c7ffe25f32afecb3aaaa7e670aR1-R3)
* [`src/pages/callback.tsx`](diffhunk://#diff-09dc989d9a1de2a2410799c328454bbc2de21f3318ec47a6e18315b9e5e26af7R1-R21): Added a new page to render the `CallbackPage` component and hide the navbar on load.

### Updates to existing components:

* [`src/components/UserNavbarItem/item.desktop.tsx`](diffhunk://#diff-47ad90b648ff92b87d9f36c9a5fc8a0f39ad643783e93d9b75da20da1c6944f4R15): Updated `SignedInActions` to use `handleLogin` from `useHandleLogin` instead of `handleClick`. [[1]](diffhunk://#diff-47ad90b648ff92b87d9f36c9a5fc8a0f39ad643783e93d9b75da20da1c6944f4R15) [[2]](diffhunk://#diff-47ad90b648ff92b87d9f36c9a5fc8a0f39ad643783e93d9b75da20da1c6944f4R66-R75)
* [`src/features/Apiexplorer/LoginDialog/index.tsx`](diffhunk://#diff-5c034fe2ccf598f7f5d2c87fca0f074989efcaf0ae58a75594fdd9823b5fb860R7): Modified `LoginDialog` to use `handleLogin` from `useHandleLogin` for the login button. [[1]](diffhunk://#diff-5c034fe2ccf598f7f5d2c87fca0f074989efcaf0ae58a75594fdd9823b5fb860R7) [[2]](diffhunk://#diff-5c034fe2ccf598f7f5d2c87fca0f074989efcaf0ae58a75594fdd9823b5fb860R29-R32) [[3]](diffhunk://#diff-5c034fe2ccf598f7f5d2c87fca0f074989efcaf0ae58a75594fdd9823b5fb860L60-R65)
* [`src/features/Login/Login.tsx`](diffhunk://#diff-ee853fb866d68d714ae9530ec456aa024accc5ecf0e3e333ed8dfcac8a97565dR8): Updated `Login` component to use `handleLogin` from `useHandleLogin` for the login button. [[1]](diffhunk://#diff-ee853fb866d68d714ae9530ec456aa024accc5ecf0e3e333ed8dfcac8a97565dR8) [[2]](diffhunk://#diff-ee853fb866d68d714ae9530ec456aa024accc5ecf0e3e333ed8dfcac8a97565dR19-R23) [[3]](diffhunk://#diff-ee853fb866d68d714ae9530ec456aa024accc5ecf0e3e333ed8dfcac8a97565dL29-R35)
* `src/features/dashboard/dashboard.tsx` and `src/pages/auth.tsx`: Wrapped `Login` component with `BrowserOnly` to ensure it only renders on the client side. [[1]](diffhunk://#diff-2bacbec4191178fbf30071ffce48c5e21846c22bb0880212f72a1115444b6929R6) [[2]](diffhunk://#diff-2bacbec4191178fbf30071ffce48c5e21846c22bb0880212f72a1115444b6929L19-R20) [[3]](diffhunk://#diff-f41120866cc9359e8272b3546f6e6daa3da48caaeadf7ef37392fe0e62df8becR8) [[4]](diffhunk://#diff-f41120866cc9359e8272b3546f6e6daa3da48caaeadf7ef37392fe0e62df8becL34-R35)

### Dependency updates:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L31-R31): Updated `@deriv-com/auth-client` dependency from version `^1.0.29` to `^1.2.15`.

### Utility function enhancements:

* `src/utils/index.ts` and `src/utils/__tests__/utils.test.ts`: Added `transformAccountsFromResponseBody` function to transform account response data into an array of objects, and included corresponding tests. [[1]](diffhunk://#diff-6147e3c1761df71fd40952dbcbac388a45f80b8c318f367ef41048351a2c0c99R109-R132) [[2]](diffhunk://#diff-6147e3c1761df71fd40952dbcbac388a45f80b8c318f367ef41048351a2c0c99L129-R153) [[3]](diffhunk://#diff-dae6f72a414341bc0a62778146f7f008c64e64fa48321ecabd680de65d68c72eR5) [[4]](diffhunk://#diff-dae6f72a414341bc0a62778146f7f008c64e64fa48321ecabd680de65d68c72eR110-R140)

### CODE REVIEW

File: src/features/Apiexplorer/LoginDialog/index.tsx [hunk 1]
```diff
@@ -57,7 +62,7 @@ export const LoginDialog = ({ setToggleModal }: TLoginDialog) => {
               <Button color='tertiary' onClick={handleSignUp} className={styles.btn}>
                 <Translate>Sign up</Translate>
               </Button>
-              <Button color='primary' onClick={handleClick} className={styles.btn}>
+              <Button color='primary' onClick={handleLogin} className={styles.btn}>
```
Comment (amam-deriv) [inline src/features/Apiexplorer/LoginDialog/index.tsx:65]: [Q] do we not have feature flag to toggle between phase 1 and 2?

File: src/features/dashboard/dashboard.tsx [hunk 2]
```diff
@@ -16,7 +17,7 @@ const Dashboard = () => {
   }, [setIsDashboard]);
 
   if (is_logged_in) return <ManageDashboard />;
-  return <Login />;
+  return <BrowserOnly>{() => <Login />}</BrowserOnly>;
```
Comment (amam-deriv) [inline src/features/dashboard/dashboard.tsx:20]: [Q] is this flow update expected?

File: src/features/Apiexplorer/LoginDialog/index.tsx (hunk 1 above)
Comment (thisyahlen-deriv) [reply src/features/Apiexplorer/LoginDialog/index.tsx:65]: Handled in the useHandleLogin hook 🙏

File: src/features/dashboard/dashboard.tsx (hunk 2 above)
Comment (thisyahlen-deriv) [reply src/features/dashboard/dashboard.tsx:20]: Yeap cause we are using the useHandleLogin hook inside the Login component

File: src/pages/callback.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+const Callback = () => {
+  useEffect(() => {
+    const navbar = document.getElementsByClassName('navbar navbar--fixed-top')[0] as HTMLElement;
+    if (navbar) {
+      navbar.style.display = 'none';
+    }
+  }, []);
+  return (
+    <Layout title='Callback' description=''>
```
Comment (amam-deriv) [inline src/pages/callback.tsx:13]: [R] we should make this page non-crawlable on SEO side

**Review by amam-deriv — CHANGES_REQUESTED** [review]
approach lgtm, however, 1 missing scenario

File: src/pages/callback.tsx (hunk 3 above)
Comment (thisyahlen-deriv) [reply src/pages/callback.tsx:22]: done

**Review by shafin-deriv — APPROVED** [review]

**Review by hasan-deriv — APPROVED** [review]

**Review by amam-deriv — APPROVED** [review]
[M] let's check how do we generate sitemap here and add filter for callback

**Review by ahmed-deriv — APPROVED** [review]

**Pull request merged**